controllers/gitopsservice_controller.go

```diff
@@ -330,6 +331,9 @@ func (r *ReconcileGitopsService) reconcileDefaultArgoCDInstance(instance *pipeli
 			NodeSelector: common.InfraNodeSelector(),
 		}
 	}
+	if len(instance.Spec.NodeSelector) > 0 {
```

Comment:
I was wondering, how does this behave in combination with `runOnInfra` set to true, or with `tolerations` set to any value.

To me it looks like they overwrite each other. For example, if `toleations` and `nodeSelector` is set, `nodeSelector` will be overwritten by `tolerations` below.

Is this expected?

Reply:
These are actually working on the nodePlacement property in the argocd CR, so there is a check to see if that property exists in the CR already, if it does then it just appends or else creates. So they won't overwrite each other when multiple options are set.